src/server/webkit.ts

```diff
@@ -88,143 +87,112 @@ export class WebKit extends BrowserTypeBase {
   }
 }
 
-function wrapTransportWithWebSocket(transport: ConnectionTransport, logger: InnerLogger, port: number): WebSocketWrapper {
-  const server = new ws.Server({ port });
-  const guid = helper.guid();
-  const idMixer = new SequenceNumberMixer<{id: number, socket: ws}>();
+type PageProxyData = {
+  socket: ws,
+};
+
+function startWebSocketServer(transport: ConnectionTransport, logger: InnerLogger, port: number): WebSocketServer {
   const pendingBrowserContextCreations = new Set<number>();
   const pendingBrowserContextDeletions = new Map<number, string>();
   const browserContextIds = new Map<string, ws>();
-  const pageProxyIds = new Map<string, ws>();
-  const sockets = new Set<ws>();
+  const pageProxyIds = new Map<string, PageProxyData>();
 
-  transport.onmessage = message => {
-    if (typeof message.id === 'number') {
-      if (message.id === -9999)
-        return;
-      // Process command response.
-      const value = idMixer.take(message.id);
-      if (!value)
+  function removePageProxy(pageProxyId: string): PageProxyData | undefined {
+    const data = pageProxyIds.get(pageProxyId);
+    if (!data)
+      return;
+    pageProxyIds.delete(pageProxyId);
+    return data;
+  }
+
+  const server = new WebSocketServer(transport, logger, port, {
+    onBrowserResponse(seqNum: number, source: ws, message: ProtocolResponse) {
```

Comment:
nit: I'd avoid seqNum here and use "id" to be in protocol terms.

Reply:
Tried that and immediately got confused. I'd like renamed ids to be consistently called seqNums